controlled-vocabulary/src/main/java/org/uniprot/cv/xdb/UniProtDatabaseTypes.java

```diff
@@ -112,4 +110,19 @@ private void init() {
                 types.stream()
                         .collect(Collectors.toMap(UniProtDatabaseDetail::getName, val -> val));
     }
+
+    String getSourceAsString() {
+        try {
+            String location = getDrDatabaseTypesLocation();
+            if (location != null && !location.isBlank()) {
```

Comment:
We have Utils class in core-common  with lots of helper method to check null and empty. You may use that.